### PR TITLE
feat: Hands off workflows 

### PR DESCRIPTION
**Pull request by Maed223**
### Related issue

[Per the summer house keeping item](https://github.com/orgs/hashicorp/projects/168/views/15?pane=issue&itemId=34263643)

### Description

Adds workflows for Dependabot, automerge, and auto-approve. Additionally, workflows have been added snapshot testing (to validate against a regression from dependency updates).

#### Snapshot Testing
Snapshot testing (of synthed stacks) is used to detect any drift from the original configuration between dependency updates. As such any version info coming from the synthed stacks are removed prior to to validating against the snapshot in order to avoid CI failing for dependabot updates.

### Checklist

- [ ] I have updated the PR title to match [CDKTF's style guide](https://github.com/cdktf/.github/blob/main/CONTRIBUTING.md#pull-requests-1)
- [ ] I have run the linter on my code locally
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation if applicable
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works if applicable
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/automerge.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    permissions:
+      contents: read
+    if: contains(github.event.pull_request.labels.*.name, 'automerge') && github.event.pull_request.draft == false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - name: Turn on automerge for this PR
+        run: gh pr merge --auto --squash ${{ github.event.pull_request.number }}
+        env:
+          GH_TOKEN: ${{ secrets.TEAM_TF_CDK }}
```
Comment (ansgarm) [inline .github/workflows/automerge.yml:23]: Is this secret set already?

File: test/test-against-regression.test.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    it("has no change in FrontendStack Configuration", () => {
+        const app = Testing.app();
+        const stack = new FrontendStack(app, "test-frontend", {
+            environment: "development",
+            user: "regression-test",
+            apiEndpoint: "N/A",
+        });
+        const stackSynth = JSON.parse(Testing.synth(stack));
+        delete stackSynth["terraform"]
+        expect(JSON.stringify(stackSynth, null, 2)).toMatchSnapshot();
```
Comment (ansgarm) [inline test/test-against-regression.test.ts:15]: Is there a reason to stringify the stack? Snapshots should be able to compare objects as well (and print a nice diff if they differ) – Feel free to ignore, I'm just curious :)

**Review by ansgarm — APPROVED** [review]
Nice work, @Maed223 💯
I've got only two non-blocking small questions.

File: .github/workflows/automerge.yml (hunk 1 above)
Comment (Maed223) [reply .github/workflows/automerge.yml:23]: [Yup 👍](https://github.com/cdktf/cdktf-integration-serverless-example/settings/secrets/actions)

File: test/test-against-regression.test.ts (hunk 2 above)
Comment (Maed223) [reply test/test-against-regression.test.ts:15]: For some reason that had not occurred to me! Good to know 😁

**Pull request merged**